gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -717,3 +729,280 @@ message TranslatedString {
+  // Identifies the new serviced stop (defined in the RT-GTFS).
+  optional string stop_id = 6;
+
+  // Valid options for pickup_type:
+  enum PickupType {
+    REGULAR_PICKUP = 0;
+    NO_PICKUP = 1;
+    MUST_PHONE_AGENCY_PICKUP = 2;
+    MUST_ASK_DRIVER_PICKUP = 3;
+  }
+
+  // Valid options for drop_off_type:
+  enum DropOffType {
+    REGULAR_DROP_OFF = 0;
+    NO_DROP_OFF = 1;
+    MUST_PHONE_AGENCY_DROP_OFF = 2;
+    MUST_ASK_DRIVER_DROP_OFF = 3;
+  }
+}
+
+// Describes new trips.
+message Trip {
+  // Identifier of the trip. Must be different than any trip_id defined in the GTFS.
+  required string trip_id = 1;
+  // Identifies a route.
+  optional string route_id = 2;
+  // Text that appears on signage identifying the trip's destination to riders.
+  required TranslatedString trip_headsign = 3;
+  // Public facing text used to identify the trip to riders, for instance, to identify train numbers for commuter rail trip
+  required string trip_short_name = 4;
+  // Indicates the direction of travel for a trip.
+  optional uint32 direction_id = 5;
+  // Identifies the block to which the trip belongs.
+  optional string block_id = 6;
+  // Identifies a geospatial shape describing the vehicle travel path for a trip.
+  optional string shape_id = 7;
+  // Indicates wheelchair accessibility.
+  optional WheelchairAccessibleStatus wheelchair_accessible = 8 [default = UNKNOWN_WHEELCHAIR_ACCESSIBILITY];
+  // Indicates whether bikes are allowed or not.
+  optional BikesAllowedStatus biked_allowed = 9 [default = UNKNOWN_BIKES_ALLOWANCE];
+  //  Defines which trips are being replaced by this trip.
+  repeated string replaces_trip_id = 10;
+  // The arrival and departure times for this trip. Entities must be ordered by stop_sequence.
+  repeated StopTime stop_time = 11;
+
+  // Options for wheelchair_accessible:
+  // 0 - No accessibility information for the trip.
+  // 1 - Vehicle being used on this particular trip can accommodate at least one rider in a wheelchair.
+  // 2 - No riders in wheelchairs can be accommodated on this trip.
+  enum WheelchairAccessibleStatus {
+    UNKNOWN_WHEELCHAIR_ACCESSIBILITY = 0;
+    WHEELCHAIR_ACCESSIBLE = 1;
+    NOT_WHEELCHAIR_ACCESSIBLE = 2;
+  }
+
+  // Options for biked_allowed:
+  // 0 - No bike information for the trip.
+  // 1 - Vehicle being used on this particular trip can accommodate at least one bicycle.
+  // 2 - No bicycles are allowed on this trip.
+  enum BikesAllowedStatus {
+    UNKNOWN_BIKES_ALLOWANCE = 0;
+    BIKES_ALLOWED = 1;
+    BIKES_NOT_ALLOWED = 2;
+  }
+}
+
+// Describes the times that a vehicle arrives at and departs from stops for each updated trip
+message StopTime {
+  // Order of stops for a particular trip. The values must increase along the trip but do not need to be consecutive.
+  required uint32 stop_sequence = 1;
+  // Arrival time at a specific stop for a specific trip on a route.
+  // This field must be in the format of HH:MM:SS.
+  required string arrival_time = 2;
```

Comment:
Here, the proposal specifies epoch (therefore uint64) but comments propose HH:MM:SS format (see https://docs.google.com/document/d/1oPfQ6Xvui0g3xiy1pNiyu5cWFLhvrTVyvYsMVKGyGWM/edit?disco=AAAAD25IX2g).

Just making sure that we should keep the HH:MM:SS format

Reply:
Yes, we should keep "HH:MM:SS". Scott and @timMillet are working to update the spec document to reflect "HH:MM:SS".

Reply:
How do you represent the dates the trip is running? Maybe a `start_date` on `Trip`?

Reply:
For anyone looking at this thread but hasn't looked at the Google Doc - one reason for preferring "HH:MM:SS" is to be able to specify frequency-based trips as well. To support this, there will be more fields added soon to mirror GTFS static CSV frequencies.txt (but only for exact_times=0 trips).